coderd/database/migrations/000266_cryptokey_features.up.sql

```diff
@@ -0,0 +1,18 @@
+-- Create a new enum type with the desired values
+CREATE TYPE new_crypto_key_feature AS ENUM (
+    'workspace_apps_token',
+    'workspace_apps_api_key',
+    'oidc_convert',
+    'tailnet_resume'
+);
+
+DELETE FROM crypto_keys WHERE feature = 'workspace_apps';
+
+-- Drop the old type and rename the new one
+ALTER TABLE crypto_keys
+    ALTER COLUMN feature TYPE new_crypto_key_feature
+    USING (feature::text::new_crypto_key_feature);
+
+DROP TYPE crypto_key_feature;
+
+ALTER TYPE new_crypto_key_feature RENAME TO crypto_key_feature;
```

Comment:
Is this a separate issue from [this comment](https://github.com/coder/coder/pull/15066#discussion_r1804521384). The only time redirecting would be helpful is if you got caught mid-upgrade right? If you're in a scenario where you're partially upgraded (say the `wsproxy` is using old-style and `coderd` is using new-style) redirecting to reauth won't help and in fact could make it more confusing because you're going to hit `too many redirects`

Reply:
So just to summarize the consequences:
- `oidc_convert`: Will throw an error asking the user to retry. This should be the rarest of them all and should be a one-off error
- `workspace_apps_api_key`: Will throw an error asking the user to retry. Should be the second rarest since most users will already be authenticated.
- `workspace_apps_token`: Will silently discard token and issue a new token (presuming the api key is still valid)
- `tailnet_resume`: Will ignore the token and generate a new peer id

I've added tests for each case thus the balloon in PR size 😞